hw/pci-bridge/cxl_socket_transport.c

```diff
@@ -354,31 +351,44 @@ bool send_cxl_io_mem_write(int socket_fd, hwaddr hpa, uint64_t val, int size,
 
     assert(size % 4 == 0); // Ensure size is dword aligned
 
-    cxl_io_mem_wr_packet_t packet;
-
-    packet.system_header.payload_type = CXL_IO;
-    packet.system_header.payload_length = sizeof(packet);
-
-    packet.cxl_io_header.fmt_type = MWR_64B;
-    uint16_t hdr_length = round_up_to_nearest_dword(size);
-    packet.cxl_io_header.length_upper = EXTRACT_UPPER_2(hdr_length);
-    packet.cxl_io_header.length_lower = EXTRACT_LOWER_8(hdr_length);
-
-    packet.mreq_header.req_id = 0;
-    packet.mreq_header.tag = *tag;
-
-    hpa = htonll(hpa);
+    cxl_io_mem_base_packet_t *base;
+    cxl_io_mem_wr_packet_32b_t packet_32;
+    cxl_io_mem_wr_packet_64b_t packet_64;
+    bool successful;
+
+    uint16_t hdr_length = round_up_to_nearest_dword(size) / 4;
+    size_t payload_length = 0;
+
+    if (size == 8) {
+        assert(val < UINT64_MAX);
```

Comment:
Thinking of it, this is actually a meaningless check (i.e. always true) because (UINT64_MAX + 1) is not greater than UINT64_MAX on a uint64_t variable due to overflow.

Remove.